osmankh/EMS
# [FEAT] [ENDPOINT] `POST /api/expenses` to add expense

## Changes

- Added `Post /api/expenses`
- Exceptions
- Exception Listener
- Form Validator
- Refactor unit tests.
- Added Mapper
- Added Integration Test

## To Test

- run `composer test` and expect 100% success rate
- hist `POST /api/expenses` manually then call the get and assert your expense added

```curl
curl --location --request POST 'http://localhost:8000/api/expenses' \
--header 'Content-Type: application/json' \
--data-raw '{
    "description": "Test Description",
    "type": "Entertainment",
    "value": 12
}'
```
